sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -614,6 +615,15 @@ def _pardo_fn_data(transform_node, get_label):
             transform_node.inputs[0].windowing)
 
   def apply_CombineValues(self, transform, pcoll):
+    # TODO(BEAM-2937): Disable combiner lifting for fnapi. Remove this
+    # restrictions once this feature is supported in the portability framework.
+    debug_options = pcoll.pipeline._options.view_as(DebugOptions)
+    use_fn_api = (
+        debug_options.experiments and 'beam_fn_api' in debug_options.experiments
```

Comment:
I assume this is inferred when streaming is set?

Reply:
Thank you for catching this. It is inferred in the apiclient, but it is inferred after the graph was built.